src/Controller/StoreApiController.php

```diff
@@ -164,10 +165,18 @@ public function postPaymentDetails(
 
         $details = $stateData['details'];
 
-        $result = $this->paymentDetailsService->doPaymentDetails(
-            $details,
-            $paymentResponse->getOrderTransaction()
-        );
+        try {
+            $result = $this->paymentDetailsService->doPaymentDetails(
+                $details,
+                $paymentResponse->getOrderTransaction()
+            );
+        } catch (PaymentFailedException $exception) {
+            $this->logger->error(
```

Comment:
It is already logged in `src/Service/PaymentDetailsService.php` where it is thrown.

Reply:
maybe it makes sense to only log errors at the higher levels, e.g in the controller and in the public payment handler method

Reply:
On a second thought, we need to log more specific error details at the lower levels because there we have access to the specific data causing the error, like [here](https://github.com/Adyen/adyen-shopware6/blob/dae73748120dd9897edfc70a137e7757cfc88a6c/src/Service/PaymentDetailsService.php#L144)

Reply:
yes and for these we can also include the $exception message I believe